core/src/analyze.rs

```diff
@@ -15,26 +15,43 @@ pub enum AnalyzeError {
     },
 }
 
+/// This represents the finished second step in the processing pipeline.
+/// The compile time bindings have been further processed to a form that allows us to easily create
+/// the cartesian product and thereby all query variations in the next step.
 #[derive(Debug)]
 pub(crate) struct AnalyzedConditionalQueryAs {
     pub(crate) output_type: syn::Ident,
     pub(crate) query_string: syn::LitStr,
-    pub(crate) comp_time_bindings: Vec<CompileTimeBinding>,
+    pub(crate) compile_time_bindings: Vec<CompileTimeBinding>,
 }
 
+/// This represents a single combination of a single compiletime binding of a query.
 #[derive(Debug)]
 pub(crate) struct CompileTimeBinding {
+    /// The actual expression used in the match statement.
+    /// E.g. for `match something`, this would be `something`.
     pub(crate) expression: syn::Expr,
+    /// Each entry in this Vec represents a single expanded `match` and the
+    /// binding names with the binding values from that specific arm.
+    /// (`match arm pattern`, Vec(binding_name, binding_value)`
+    /// REVIEW: How about we name this `arms`?
+    /// REVIEW: In theory, this also allow non-enum match statements and `variants`
+    /// REVIEW: might be a misleading.
     pub(crate) variants: Vec<(syn::Pat, Vec<(syn::Ident, syn::LitStr)>)>,
```

Comment:
Good point.  While writing this I don't think I was even really thinking of them as match arms anymore, but I agree that it would be a less misleading name.